docs/graph_engine/custom_decoder.rst

```diff
@@ -0,0 +1,130 @@
+**************************************************
+How to Create a Custom Decoder from Avro to Binary
+**************************************************
+
+In this example we take an Avro file as input, write a Avro to binary
+decoder and then use this decoder in to generate binaries
+that can be loaded by our graph engine.
+
+First we generate an example Avro file to use. This is a binary format with a schema to
+define the columns and their types and a binary file where each row is a record.
+This table has the columns: "src", "dst", "type", "weight", "float_feature" and "int_feature".
+Each row represents one record, either a node or an edge.
+
+In this file we have two nodes, 0 and 1, on lines 0 and 2 in which the src column gives their
+node ID and the dst is -1. We also have an edge in each direction between the two on lines 1 and 3.
+It is important that we order these rows specifically, first with a node then all of its outgoing
```

Comment:
Could you actually show how to sort it in pyspark? This is a nontrivial part.

Reply:
In order to enable pyspark avro support you need command line arg // spark conf edit

https://spark.apache.org/docs/latest/sql-data-sources-avro.html#:~:text=./bin/spark%2Dsubmit%20%2D%2Dpackages%20org.apache.spark%3Aspark%2Davro_2.12%3A3.3.0%20...

which is not possible to do in doctest. There seems to be no environment variable to handle this either: https://spark.apache.org/docs/latest/configuration.html#environment-variables

Reply:
Well, since you are not using spark anymore, we can do sorting directly on a file, right?

Reply:
We can sort while writing, the avro-python3 package does not seem to have any sorting functonality.